Bug: "File './tsconfig-base' not found" or "Cannot read file 'src/src/src/src/tsconfig-base.json'" in alpha (after ProjectService PR?)

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

I bumped TypeScript up to the latest ts-eslint alpha to try and test out the new project service. However, I went to run a "before" benchmark and noticed that ts-eslint fails to load the configuration / tsconfig we have in the repo already, even before going to change the config.

Running eslint, I get this for each file:

```
/home/jabaile/work/TypeScript/src/typingsInstallerCore/_namespaces/ts.server.ts
  0:0  error  Parsing error: File './tsconfig-base' not found

/home/jabaile/work/TypeScript/src/typingsInstallerCore/_namespaces/ts.server.typingsInstaller.ts
  0:0  error  Parsing error: File './tsconfig-base' not found

/home/jabaile/work/TypeScript/src/typingsInstallerCore/_namespaces/ts.ts
  0:0  error  Parsing error: File './tsconfig-base' not found

/home/jabaile/work/TypeScript/src/typingsInstallerCore/typingsInstaller.ts
  0:0  error  Parsing error: File './tsconfig-base' not found

/home/jabaile/work/TypeScript/src/watchGuard/watchGuard.ts
  0:0  error  Parsing error: File './tsconfig-base' not found

✖ 576 problems (576 errors, 0 warnings)
```

Modifying our tsconfig-eslint base to instead try an explicit `.json` extension nets:


```
/home/jabaile/work/TypeScript/src/typingsInstallerCore/_namespaces/ts.server.typingsInstaller.ts
  0:0  error  Parsing error: Cannot read file 'src/src/src/src/tsconfig-base.json'

/home/jabaile/work/TypeScript/src/typingsInstallerCore/_namespaces/ts.ts
  0:0  error  Parsing error: Cannot read file 'src/src/src/src/tsconfig-base.json'

/home/jabaile/work/TypeScript/src/typingsInstallerCore/typingsInstaller.ts
  0:0  error  Parsing error: Cannot read file 'src/src/src/src/tsconfig-base.json'

/home/jabaile/work/TypeScript/src/watchGuard/watchGuard.ts
  0:0  error  Parsing error: Cannot read file 'src/src/src/src/tsconfig-base.json'

✖ 576 problems (576 errors, 0 warnings)
```

I suspect that maybe something regressed in said ProjectService change, but I can't be sure.

### Reproduction Repository Link

https://github.com/jakebailey/TypeScript/tree/eslint-v6-alpha

### Repro Steps

1) clone the repo
2) `npm ci`
3) `npx eslint .`


### Versions

<!-- Fill in the relevant versions and delete the others -->
| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `6.0.1-alpha.28` |
| `@typescript-eslint/parser`            | `6.0.1-alpha.28` |
| `@typescript-eslint/scope-manager`     | `6.0.1-alpha.28` |
| `@typescript-eslint/typescript-estree` | `6.0.1-alpha.28` |
| `@typescript-eslint/type-utils`        | `6.0.1-alpha.28` |
| `@typescript-eslint/utils`             | `6.0.1-alpha.28` |
| `TypeScript`                           | `5.1.6` |
| `ESLint`                               | `8.45.0` |
| `node`                                 | `20.4.0` |

@jakebailey what version of npm are you using? On 9.6.6, I get:

```plaintext
$ npm ci
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @typescript-eslint/eslint-plugin@6.0.1-alpha.28
npm ERR! Found: @typescript-eslint/parser@6.0.1-alpha.28
npm ERR! node_modules/@typescript-eslint/parser
npm ERR!   dev @typescript-eslint/parser@"6.0.1-alpha.28" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @typescript-eslint/parser@"^6.0.0 || ^6.0.0-alpha" from @typescript-eslint/eslint-plugin@6.0.1-alpha.28
npm ERR! node_modules/@typescript-eslint/eslint-plugin
npm ERR!   dev @typescript-eslint/eslint-plugin@"6.0.1-alpha.28" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: @typescript-eslint/parser@6.0.0
npm ERR! node_modules/@typescript-eslint/parser
npm ERR!   peer @typescript-eslint/parser@"^6.0.0 || ^6.0.0-alpha" from @typescript-eslint/eslint-plugin@6.0.1-alpha.28
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"6.0.1-alpha.28" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/josh/.npm/_logs/2023-07-17T13_47_07_103Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/josh/.npm/_logs/2023-07-17T13_47_07_103Z-debug-0.log
```

I can repro your issue with `npm ci --force` though.

--- a/packages/typescript-estree/src/create-program/getWatchProgramsForProjects.ts
+++ b/packages/typescript-estree/src/create-program/getWatchProgramsForProjects.ts
@@ -261,10 +261,7 @@ function createWatchProgram(
   const watchCompilerHost = ts.createWatchCompilerHost(
     tsconfigPath,
     createDefaultCompilerOptionsFromExtra(parseSettings),
-    {
-      ...ts.sys,
-      getCurrentDirectory: () => parseSettings.tsconfigRootDir,
-    },
+    ts.sys,
     ts.createAbstractBuilder,
     diagnosticReporter,
     // TODO: file issue on TypeScript to suggest making optional?

